measure

```diff
@@ -0,0 +1,181 @@
+
+        comps = app.get('components')
+        assert comps and isinstance(comps, dict), "Config located at {} was malformed; `{}`'s `application` section must include `components` dictionary, found: {}".format(CONFIG_PATH, DRIVER_NAME, comps.__class__.__name__)
+
+        for c_name, c_data in comps.items():
+            assert isinstance(c_data, dict),  "Config located at {} was malformed; `{}` application component `{}` value must be a dictionary, found: {}".format(CONFIG_PATH, DRIVER_NAME, c_name, c_data.__class__.__name__)
+            settings = c_data.get('settings')
+            assert isinstance(settings, dict), "Config located at {} was malformed; `{}` application component `{}` must contain `settings` dict, found: {}".format(CONFIG_PATH, DRIVER_NAME, c_name, c_data.__class__.__name__)
+            assert settings, "Config located at {} was malformed; `{}` application component `{}` provided no settings".format(CONFIG_PATH, DRIVER_NAME, c_name)
+
+            for s_name, s_data in settings.items():
+                for s in REQUIRED_SETTINGS_FIELDS:
+                    assert s in s_data, \
+                        "missing required key '{}' from driver configuration for setting {}, component {}".format(
+                            s, s_name, c_name)
+
+        command = config.get('command')
+        assert isinstance(command, str), 'The `command` attribute in `batch` section must be a string, found {}'.format(type(command).__name__)
+
+        ed = config.get('expected_duration')
+        if ed:
+            assert (isinstance(ed, int) or isinstance(ed, float)) and ed > 0, 'The `expected_duration` attribute in `batch` section must be a numeric type greater than zero when specified. Found: {} `{}`'.format(type(ed).__name__, ed)
+
+        metrics = config.get('metrics')
+        if metrics:
+            assert isinstance(metrics, dict), 'The `metrics` attribute in `batch` section must be a dict when provided, found {}'.format(type(metrics).__name__)
+
+            for m_name, m_data in metrics.items():
+                out_reg = m_data.get('output_regex')
+                assert isinstance(out_reg, str), 'The `output_regex` attribute is required in metric `{}` under `batch` section and must be a string, found {}'.format(m_name, type(metrics).__name__)
+                try:
+                    re.compile(out_reg)
+                except re.error as e:
+                    raise Exception('Failed to compile the `output_regex` of metric `{}` in `batch` section'.format(m_name)) from e
+
+    def describe(self):
+        metrics = { 'total_runtime': { 'unit': 'seconds' } }
+        for m_name, m_data in self.config.get('metrics', {}).items():
+            metrics[m_name] = {}
+            unit = m_data.get('unit')
+            if unit:
+                metrics[m_name]['unit'] = unit
+
+        return metrics
+
+    def _print_progress_wrapper(self):
```

Comment:
Does this need to be a wrapper? Can it simply overwrite the parent's method

Reply:
Given there is no need modify the parent functionality beyond setting the progress percentage, I found it simplest to call the parent's method from the wrapper here

https://github.com/opsani/servo-batch/blob/111f6dca937517313110caa7bc65a0dd13825dfa/measure#L113

And then override the print_progress_message of the driver class here

https://github.com/opsani/servo-batch/blob/111f6dca937517313110caa7bc65a0dd13825dfa/measure#L34

If I were to overwrite the entire function, I would need to redundantly implement all the code contained within this method

https://github.com/opsani/servo/blob/0947d14f2b23677b74960b3e76df80ba2555353c/measure.py#L151